kernel/shared/src/main/scala/cats/effect/kernel/Poll.scala

```diff
@@ -18,4 +18,5 @@ package cats.effect.kernel
 
 import cats.~>
 
+/** @see https://typelevel.org/cats-effect/docs/typeclasses/monadcancel */
```

Comment:
Maybe just directly link to `MonadCancel` scaladoc, which is quite comprehensive?

```suggestion
/** @see [[MonadCancel]] */
```